FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,319 @@
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.AnalyticsConnector.Common.Logging;
+using Microsoft.Health.AnalyticsConnector.Core.Exceptions;
+using Microsoft.Health.AnalyticsConnector.Core.Extensions;
+using Microsoft.Health.AnalyticsConnector.Core.Fhir;
+using Microsoft.Health.AnalyticsConnector.Core.Jobs.Models;
+using Microsoft.Health.AnalyticsConnector.DataClient;
+using Microsoft.Health.AnalyticsConnector.DataClient.Api.Fhir;
+using Microsoft.Health.AnalyticsConnector.DataClient.Extensions;
+using Microsoft.Health.AnalyticsConnector.DataClient.Models.FhirApiOption;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Polly.Timeout;
+
+namespace Microsoft.Health.AnalyticsConnector.Core.Jobs
+{
+    public class FhirToDataLakeProcessingJobSpliter
+    {
+        private readonly IApiDataClient _dataClient;
+        private readonly IDiagnosticLogger _diagnosticLogger;
+        private readonly ILogger<FhirToDataLakeProcessingJobSpliter> _logger;
+        private const string CountApiParameter = "count";
+        private const string TotalCountFieldKey = "total";
+        private const string LastUpdatedApiParameter = "_lastUpdated";
+        private const string LastUpdatedApiParameterDesc = "-_lastUpdated";
+        private const int TimeoutSecondsGetResourceCount = 90;
+
+        public FhirToDataLakeProcessingJobSpliter(
+            IApiDataClient dataClient,
+            IDiagnosticLogger diagnosticLogger,
+            ILogger<FhirToDataLakeProcessingJobSpliter> logger)
+        {
+            _dataClient = EnsureArg.IsNotNull(dataClient, nameof(dataClient));
+            _diagnosticLogger = EnsureArg.IsNotNull(diagnosticLogger, nameof(diagnosticLogger));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public int LowBoundOfProcessingJobResourceCount { get; set; } = JobConfigurationConstants.LowBoundOfProcessingJobResourceCount;
+
+        public int HighBoundOfProcessingJobResourceCount { get; set; } = JobConfigurationConstants.HighBoundOfProcessingJobResourceCount;
+
+        public async IAsyncEnumerable<SubJobInfo> SplitJobAsync(string resourceType, DateTimeOffset? startTime, DateTimeOffset endTime, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            var totalCount = await GetResourceCountAsync(resourceType, startTime, endTime, cancellationToken);
+
+            if (totalCount <= HighBoundOfProcessingJobResourceCount)
+            {
+                // Return job with total count lower than high bound.
+                _logger.LogInformation($"Generate one {resourceType} job with {totalCount} count.");
```

Comment:
nit, we should make log information more verbose. might be "Split one sub job with {count} resources from {start} to {end} for resource type {}."